WordPress/src/main/java/org/wordpress/android/ui/sitecreation/creation/NewSiteCreationServiceState.kt

```diff
@@ -0,0 +1,37 @@
+package org.wordpress.android.ui.sitecreation.creation
+
+import org.wordpress.android.util.AutoForeground
+
+data class NewSiteCreationServiceState internal constructor(
+    val step: NewSiteCreationStep,
+    val payload: Any? = null
+) : AutoForeground.ServiceState {
+    override fun isIdle(): Boolean {
+        return step == NewSiteCreationStep.IDLE
+    }
+
+    override fun isInProgress(): Boolean {
+        return step != NewSiteCreationStep.IDLE && !isTerminal
+    }
+
+    override fun isError(): Boolean {
+        return step == NewSiteCreationStep.FAILURE
+    }
+
+    override fun isTerminal(): Boolean {
+        return step == NewSiteCreationStep.SUCCESS || isError
+    }
+
+    override fun getStepName(): String {
+        return step.name
+    }
+
+    enum class NewSiteCreationStep(val progressPercent: Int = 0) {
+        IDLE,
+        CREATE_SITE(25),
```

Comment:
Constant added in [1601dbd](https://github.com/wordpress-mobile/WordPress-Android/pull/8795/commits/1601dbd443bb980941411762c7e37b1e25cb82e2).

>Better yet, should the percentages be dynamically calculated?

I've discussed this with Sylvester and we've decided having a static value is enough for v1. 

> I am guessing it's supposed to be 100 right now, since it's the only step.

It's the percentage on the progress bar when the Site Creation is in progress. Since we want to indicate something is happening but it's not done yet, we've went for 25%.

@SylvesterWilmott I guess Oguz has a point -> should we completely remove the progress bar for v1 perhaps? As it seems quite useless.

Reply:
I mean, we could switch for an indeterminate progress bar just to show something is happening.

Reply:
I've make the progress indeterminate in Removed progress in [1e30994](https://github.com/wordpress-mobile/WordPress-Android/pull/8795/commits/1e309948aa1cd409ae1287c3317b36751fd3ca56)